src/Tizen.Applications.ComponentBased.Port/Tizen.Applications.ComponentBased.Port/ComponentPort.cs

```diff
@@ -177,34 +251,69 @@ public object SendSync(string endpoint, int timeout, object request)
             {
                 object response = FromParcel(resParcel);
                 return response;
-            }            
+            }
         }
 
         /// <summary>
-        /// Abstract method for receiving a request event.
+        /// Sends the request data and receives the reply data asynchronously.
         /// </summary>
-        /// <param name="sender">The name of the sender</param>
-        /// <param name="request">The serializable data</param>
-        /// <since_tizen> 9 </since_tizen>
-        protected abstract void OnRequestEvent(string sender, object request);
+        /// <exception cref="ArgumentException">Thrown when the argument is invalid.</exception>
+        /// <exception cref="OutOfMemoryException">Thrown when the memory is insufficient.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when because of permission denied.</exception>
+        /// <exception cref="global::System.IO.IOException">Thrown when because of I/O error.</exception>
+        /// <param name="endpoint">The name of the endpoint</param>
+        /// <param name="timeout">The timeout in milliseconds, -1 to use the default timeout</param>
+        /// <param name="request">The serializable data to send</param>
+        /// <returns>The received serializable data</returns>
+        /// /// <since_tizen> 9 </since_tizen>
+        public async Task<object> SendAndReceiveAsync(string endpoint, int timeout, object request)
+        {
+            try
+            {
+                return await Task.Run(() => SendAndReceive(endpoint, timeout, request)).ConfigureAwait(false);
+            }
+            catch
+            {
+                throw;
+            }
+        }
 
         /// <summary>
-        /// Abstract method for receiving a synchronous request event.
+        /// Occurs whenever the request is received.
         /// </summary>
-        /// <param name="sender">The name of the sender</param>
-        /// <param name="request">The serializable data</param>
-        /// <returns>The serializable data</returns>
+        /// <remarks>
+        /// If the reply is requested, RequestEventArgs.Request should be set.
+        /// </remarks>
         /// <since_tizen> 9 </since_tizen>
-        protected abstract object OnSyncRequestEvent(string sender, object request);
+        public event EventHandler<RequestEventArgs> RequestReceived
```

Comment:
You can use auto generated event
```suggestion
        public event EventHandler<RequestEventArgs> RequestReceived;
```

Reply:
Done